File: Sources/CountriesFlagsObjC/source/Countries/UIImage+Togo.m
//
//  UIImage+Togo.m
//  CountriesFlags
//

#import "Countries/UIImage+Togo.h"

#import "Extensions/UIBezierPath+Star.h"

@implementation UIImage (Togo)

+ (UIImage * _Nullable)flagTogoWithSize:(CGSize)size {
    UIColor* colorRed = [UIColor colorWithRed:210.0f/255.0f
                                        green:16.0f/255.0f
                                         blue:52.0f/255.0f
                                        alpha:1.0f];
    UIColor* colorWhite = [UIColor whiteColor];
    UIColor* colorGreen = [UIColor colorWithRed:0.0f
                                          green:106.0f/255.0f
                                           blue:74.0f/255.0f
                                          alpha:1.0f];
    UIColor* colorYellow = [UIColor colorWithRed:1.0f
                                           green:206.0f/255.0f
                                            blue:0.0f
                                           alpha:1.0f];
    
    UIGraphicsBeginImageContextWithOptions(size, YES, 0.0f);
    
    [colorGreen setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorYellow setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 5.0f * 3.0f, size.width, size.height / 5.0f));
    UIRectFill(CGRectMake(0.0f, size.height / 5.0f, size.width, size.height / 5.0f));
    
    [colorRed setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.height / 5.0f * 3.0f, size.height / 5.0f * 3.0f));
    
    [colorWhite setFill];
    [[UIBezierPath starAtPoint:CGPointMake(size.height / 5.0f * 1.5f, size.height / 5.0f * 1.5f)
                        radius:size.height / 6.0f
                          type:FlagStarTypeFivePointed] fill];
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image;
}

@end

File: Sources/CountriesFlags/Countries/UIImage+Togo.swift
//
//  UIImage+Togo.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Togo "wikipedia"
     
     Generates the flag of Togo.
     For more information see [wikipedia].
     
     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Togo.
     */
    internal static func flagTogo(with size: CGSize) -> UIImage? {
        let colorRed = UIColor(red: 210.0 / 255.0,
                               green: 16.0 / 255.0,
                               blue: 52.0 / 255.0,
                               alpha: 1.0)
        let colorWhite = UIColor.white
        let colorGreen = UIColor(red: 0.0,
                                 green: 106.0 / 255.0,
                                 blue: 74.0 / 255.0,
                                 alpha: 1.0)
        let colorYellow = UIColor(red: 1.0,
                                  green: 206.0 / 255.0,
                                  blue: 0.0,
                                  alpha: 1.0)

        UIGraphicsBeginImageContextWithOptions(size, false, 0.0)

        colorGreen.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorYellow.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0, y: size.height / 5.0 * 3.0),
                          size: CGSize(width: size.width, height: size.height / 5.0)))
        UIRectFill(CGRect(origin: CGPoint(x: 0, y: size.height / 5.0),
                          size: CGSize(width: size.width, height: size.height / 5.0)))

        colorRed.setFill()
        UIRectFill(CGRect(origin: .zero,
                          size: CGSize(width: size.height / 5.0 * 3.0, height: size.height / 5.0 * 3.0)))

        colorWhite.setFill()
        UIBezierPath.star(at: CGPoint(x: size.height / 5.0 * 1.5, y: size.height / 5.0 * 1.5),
                          radius: size.height / 6.0).fill()

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image
    }
}

#endif
